packages/scalecube-microservice/src/helpers/validation.ts

```diff
@@ -0,0 +1,95 @@
+import * as check from './check';
+import {
+  SEED_ADDRESS_IS_NOT_STRING,
+  SERVICES_IS_NOT_ARRAY,
+  SERVICE_IS_NOT_OBJECT,
+  MICROSERVICE_OPTIONS_IS_NOT_OBJECT,
+  SERVICE_DEFINITION_NOT_PROVIDED,
+  SERVICE_NAME_NOT_PROVIDED,
+  DEFINITION_MISSING_METHODS,
+  INVALID_METHODS,
+  MESSAGE_NOT_PROVIDED,
+  WRONG_DATA_FORMAT_IN_MESSAGE,
+  QUALIFIER_IS_NOT_STRING,
+  INVALID_QUALIFIER,
+  INVALID_MESSAGE,
+  MESSAGE_QUALIFIER_NOT_PROVIDED,
+  MESSAGE_DATA_NOT_PROVIDED,
+  getServiceNameInvalid,
+  ASYNC_MODEL_TYPES,
+  getIncorrectMethodValueError,
+  getInvalidMethodReferenceError,
+  getAsynModelNotProvidedError,
+  getInvalidAsyncModelError,
+  getInvalidServiceReferenceError,
+  getServiceReferenceNotProvidedError,
+} from './constants';
+import { getQualifier } from '../helpers/serviceData';
+import ServiceDefinition from '../api/ServiceDefinition';
+
+export const validateMicroserviceOptions = (microserviceOptions: any) => {
+  check.assertObject(microserviceOptions, MICROSERVICE_OPTIONS_IS_NOT_OBJECT);
+  const { services, seedAddress } = microserviceOptions;
+  check.assertNonEmptyString(seedAddress, SEED_ADDRESS_IS_NOT_STRING);
+  check.assertArray(services, SERVICES_IS_NOT_ARRAY);
+  services.forEach(validateService);
+};
+
+export const validateService = (service: any) => {
+  check.assertNonEmptyObject(service, SERVICE_IS_NOT_OBJECT);
+  const { definition, reference } = service;
+  check.assertDefined(definition, SERVICE_DEFINITION_NOT_PROVIDED);
+  validateServiceDefinition(definition);
+  const { serviceName } = definition;
+  check.assertDefined(reference, getServiceReferenceNotProvidedError(serviceName));
+  validateServiceReference(reference, definition);
```

Comment:
suggestion:
maybe remove this line of code, and call it from the registry files